bee-network/src/tests/common/keys_and_ids.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::PeerId;
+
+pub fn gen_constant_peer_id() -> PeerId {
+    "12D3KooWJWEKvSFbben74C7H4YtKjhPMTDxd7gP7zxWSUEeF27st".parse().unwrap()
+}
+
+#[cfg(feature = "full")]
+mod __full {
+
+    use super::*;
+    use libp2p::identity::{
+        ed25519::{Keypair, SecretKey},
+        PublicKey,
+    };
+    use std::iter::repeat;
+
+    pub fn gen_random_peer_id() -> PeerId {
+        PeerId::from_public_key(libp2p_core::PublicKey::Ed25519(Keypair::generate().public()))
+    }
+
+    pub fn gen_deterministic_peer_id(gen: impl ToString) -> PeerId {
+        let keys = gen_deterministic_keys(gen);
+        PeerId::from_public_key(PublicKey::Ed25519(keys.public()))
+    }
+
+    pub fn gen_deterministic_keys(gen: impl ToString) -> Keypair {
+        let hex_chars = ('0'..'9').chain('a'..'f').collect::<Vec<_>>();
+        let gen = gen.to_string();
+        let gen2 = gen.to_string();
+        let chars = gen2.chars();
+
+        for c in gen.chars() {
+            if !hex_chars.contains(&c) {
+                panic!("invalid generator");
```

Comment:
Function should not panic due to malformed user input unless specifically documented. Consider returning an error instead.

Reply:
it's part of the test module, so the user here is the library owner in which case panicking is fine.